User/user.md

```diff
@@ -0,0 +1,157 @@
+          "person" :
+                {
+                "names": [
+                    {
+                        "givenName": "Mohit",
+                        "familyName": "Kumar"
+                    }
+                ],
+                "gender": "M",
+                "birthdate": "1997-09-02",
+                "addresses": [
+                {
+                 "address1": "30, Vivekananda Layout, Munnekolal,Marathahalli",
+                "cityVillage": "Bengaluru",
+                "country": "India",
+                "postalCode": "560037"
+                }
+            ]
+        },
+        "systemId": "string",
+        "userProperty": {},
+        "secretQuestion" : "What is the name of your high school?"
+    }
+    ```
+### Update a user role
+
+*  Update a target user with given UUID, this method only modifies properties in the request. Returns a `404 Not Found` 
+status if user not exists. If not logged in to perform this action, a `401 Unauthorized` status returned.
+    
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *name* | `String` | Name of the user
+    *description* | `String | Description of the user
+    *username* | `String | username of the user
+    *password* | `String | password of the user
+    *person* | `String` | person resource associated with the user
+    *systemId* | `String` | a unique identifier assigned to each user
+    *roles* | `Array[] : role` | a list of roles attributed to the user
+    *userProperties* | `JSON Object`| A set of key value pairs. Used to store user specific data 
+    *secretQuestion* | `String` | A secret question chosen by the user
+   
+    ```console
+        POST /user/:target_user_uuid
+        {
+          "name": "Mohit Sharma",
+          "description": "An OpenMRS Developer."
+          "username": "batbrain7"
+          "password": "******"
+          "person" :
+                {
+                "names": [
+                    {
+                        "givenName": "Mohit",
+                        "familyName": "Sharma"
+                    }
+                ],
+                "gender": "M",
+                "birthdate": "1997-09-02",
+                "addresses": [
+                {
+                 "address1": "30, Vivekananda Layout, Munnekolal,Marathahalli",
+                "cityVillage": "Bengaluru",
+                "country": "India",
+                "postalCode": "560037"
+                }
+            ]
+        },
+        "systemId": "string",
+        "userProperty": {},
+        "secretQuestion" : "In which year did you graduate"
+    }
+    ```
+    
+### Delete a user
+
+* Delete or Void a target user by its UUID. Returns a `404 Not Found` status if user not exists.If not logged in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *purge* | `Boolean` | The resource will be voided unless purge = ‘true’
```

Comment:
Users are [retired](https://docs.openmrs.org/datamodel/openmrs-refapp-data-model-1.9.html#users.retire), not voided.